### PR TITLE
tests: ported authselect tests from downstream

### PR DESCRIPTION
**Pull request by danlavu**
- adding authselect tests
- adding test files copied from sssd 
  - conftest.py
  - data/readme.md
  - pytest.ini
  - pyproject.toml
  - requirements.txt
  - setup.cfg

### CODE REVIEW

**Comment by sidecontrol** [comment]
Continuing the conversation from the SSSD PR https://github.com/SSSD/sssd/pull/6890

@Iker > I didn't explain myself correctly. If possible, I'd like to run all test cases in all available distributions in CI. Is there anything in RHEL that does not allow us to do so?

The authselect package can no longer be uninstalled in Fedora. There is a hard dependency with sudo.  The one minimal test cases "test_authselect__when_removed_the_uninstall_is_clean", uninstalls authselect entirely. This will only work with older versions where the sudo dependency does not exist. I don't know how many customers need this, but we should still test and support it as it works this way. We can certainly tell them this is no longer supported and they need to change with the times too.

**Comment by pbrezina** [comment]
RHEL8 and RHEL9 are stuck with `1.2.x` version, how do we want to handle this? Keep old tests for RHEL8 and 9 and start using mh tests for RHEL10? Or backport it also to 1.2.x?

**Comment by pbrezina** [comment]
Is this ready for review? I think you said that you are still working on it.

**Comment by sidecontrol** [comment]
These tests should work for 1.2, I thought we discussed major releases we will use git branches, and minor releases, use pytest markers. 

I need to finish the PAM PR before this can be merged.

**Comment by pbrezina** [comment]
> These tests should work for 1.2, I thought we discussed major releases we will use git branches, and minor releases, use pytest markers.

If these will work for 1.2 then we can backport it to 1.2.x and run it for 8 and 9 from this branch. RHEL10 can use master.

> I need to finish the PAM PR before this can be merged.

Thanks, let me know when it is ready for review.

**Comment by danlavu** [comment]
@pbrezina This is ready for review. 

Also, I do not have developer access to authselect, so I'm unable to assign this issue to myself.

**Comment by pbrezina** [comment]
Hi, I invited you to the organization.

I am not sure if I will be able to review this before the Christmas break (I'll try). Can you please also make the tests run during PR CI? See SSSD's ci.yaml for reference.

**Review by pbrezina — CHANGES_REQUESTED** [review]
Hi Dan,
please, include mhc.yaml, polarion.yaml .

All tests here are for authselect, so test_authselect.py and test_authselect prefix does not make sense. It would be better to organize the test differently eg by profile.

Included test may be organized as:
* test_sssd
* test_minimal
* test_distribution (eg test_authselect__when_removed_the_uninstall_is_clean)

Tests "*enable_and_disable*" are not sufficient. Presence of specific pam module is irrelevant, what matters is functionality. You must test that the functionality is working correctly.

test_authselect__when_removed_the_uninstall_is_clean - the problem is not RHEL vs Fedora but behavior change between authselect version. This test belongs only to 1.2.x branch.

**Comment by danlavu** [comment]
@pbrezina  ack, changing a lot of things and removing the non-functional tests, and making the placeholder/sanity into functional before the next review.

**Comment by danlavu** [comment]
@pbrezina This is ready for review now. It depends on the following PRs. 

https://github.com/SSSD/sssd-test-framework/pull/86
https://github.com/SSSD/sssd-test-framework/pull/85
https://github.com/SSSD/sssd-test-framework/pull/72

File: src/tests/system/polarion.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+testcase:
+  required:
+    title:
+      transform:
+        pattern: "^(.*)$"
+        replace: "IDM-SSSD-TC: \\1"
```
Comment (pbrezina) [inline src/tests/system/polarion.yaml:6]: Do we really use SSSD for authselect?

File: src/tests/system/tests/test_minimal.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+import pytest
+from sssd_test_framework.roles.client import Client
+from sssd_test_framework.topology import KnownTopology
+
+
+@pytest.mark.importance("critical")
+@pytest.mark.ticket(bz=1654018)
+@pytest.mark.topology(KnownTopology.Client)
+def test_minimal__local_user_authentication(client: Client):
```
Comment (pbrezina) [inline src/tests/system/tests/test_minimal.py:11]: Minimal profile is no longer available in master, (RHEL-10) please,  replace it with local.

Keep minimal for 1.2.x branch that should be used for RHEL-8,9.

**Review by pbrezina — CHANGES_REQUESTED** [review]
Thanks, this is much better and I think we can go with this for now.

In long term, however, I would like to rework it to be less sssd-like and more focused on authselect. We can not test winbind or nis stuff with the sssd framework. Lots of features are shared between profiles so we could use topology parametrization here, with the topology representing a profile. We are not that much interested in the sssd providers since we rely on the posix interface.

But its good enough for the moment. Thank you.

File: src/tests/system/polarion.yaml (hunk 1 above)
Comment (danlavu) [reply src/tests/system/polarion.yaml:6]: Yes, for example, I'll have to change the work item in polarion to match the test case name, but it's currently set to. 

IDM-SSSD-TC: authselect: Authselect SSSD profile enable and disable PAM silent lastlog

File: src/tests/system/tests/test_minimal.py (hunk 2 above)
Comment (danlavu) [reply src/tests/system/tests/test_minimal.py]: I'll create another pull request for the 1.2.x branch after this one, these tests will apply to 1.2.x , so I think a backport will be fine?

File: src/tests/system/pytest.ini [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[pytest]
+addopts = --strict-markers
+testpaths = tests
+markers =
+    rhel: marks test for rhel only
+    fedora: marks test for fedora only
```
Comment (pbrezina) [inline src/tests/system/pytest.ini:6]: Unused.

**Review by pbrezina — CHANGES_REQUESTED** [review]
Hi Dan, see the inline comment. Also please, rebase it properly without the merge commit and we can push. Thank you.

**Comment by danlavu** [comment]
Done and done.

**Review by pbrezina — APPROVED** [review]

**Pull request merged**